src/api/routes/roles.php

```
@@ -46,7 +39,7 @@
         if ($person->save()) {
             return $response->withJson(['success' => true, 'msg' => gettext('The role is successfully assigned.')]);
         } else {
-            return $response->withJson(['success' => true, 'msg' => gettext('The role could not be assigned.')]);
+            return $response->withJson(['success' => false, 'msg' => gettext('The role could not be assigned.')]);
```

you need to throw an exception with the message not return false